### ISSUE TITLE
`expand_path': user lambdas.json doesn't exist

### ISSUE THREAD

**Issue by miguelmota** [jekyll/jekyll#3405]
Hey guys, I have no idea why I'm getting this error. 
Nowhere does lambdas.json exist in my repo:
```bash
$ find ./ -name "lambdas.json" 
(empty)
$ grep -nrHoEa "lambdas.json" ./
(empty)
````

I get that error when trying to start the server:

```bash
$ jekyll serve --trace --verbose 
Configuration file: /Users/moogs/Dropbox/Development/workspace/miguelmota.github.com/_config.yml
            Source: /Users/moogs/Dropbox/Development/workspace/miguelmota.github.com
       Destination: /Users/moogs/Dropbox/Development/workspace/miguelmota.github.com/_site
 Incremental build: enabled
      Generating... 
/Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll.rb:150:in `expand_path': user lambdas.json doesn't ex
ist (ArgumentError)
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll.rb:150:in `sanitized_path'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:108:in `block in in_source_dir'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:107:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:107:in `reduce'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:107:in `in_source_dir'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:177:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:180:in `block in read_directorie
s'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:176:in `read_directories'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:155:in `read'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/site.rb:53:in `process'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/command.rb:28:in `process_site'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/commands/build.rb:58:in `build'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/commands/build.rb:34:in `process'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/lib/jekyll/commands/serve.rb:26:in `block (2 levels
) in init_with_program'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/gems/jekyll-3.0.0.pre.beta1/bin/jekyll:17:in `<top (required)>'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/bin/jekyll:23:in `load'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/bin/jekyll:23:in `<main>'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/bin/ruby_executable_hooks:15:in `eval'
        from /Users/moogs/.rvm/gems/ruby-2.2.0-preview1/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by pathawks** [comment]
When I try to build you're site on my machine, I don't get *that* error, but I notice a couple others.

 * `_layouts/blog.html` Line 36 & 58 should be `{% if %}` rather than `{{ if }}`
 * You have permalinks setup as `/:categories/:title`, but you have folders containing static files that collide with those permalinks (ie: `/blog/basic-html5-audio-manipulation/`, `blog/css3-filters`, etc.)
 * Deprecation: The 'pygments' configuration option has been renamed to 'highlighter'. Please update your config file accordingly. The allowed values are 'rouge', 'pygments' or null.

When I fixed those things, the site built for me no problem.

**Comment by parkr** [comment]
Thanks, @pathawks :)

**Comment by miguelmota** [comment]
@pathawks thanks for looking into it but I made the changes locally and still getting the same error. I had to downgrade all the way down to what I had originally, `1.5.1`, to get to run.

**Comment by pathawks** [comment]
I was running the beta of Jekyll 3. Tried with `safe` mode both off and on.

How are you building the site? `jekyll serve`?

**Comment by miguelmota** [comment]
After much trial and error I found out that if I excluded the `/vendor` directory the error would go away and the site would serve and build. I'm redoing the site so I'm not too worried about what's in that directory.

```yaml
exclude: [vendor]
```

I was getting the permalink error that you mentioned because directories exist with the same category name but I didn't want to rewrite my permalinks so I simply fixed it by adding a forward slash `/` at the end of `:title`.

```yaml
permalink: /:categories/:title/
```

No more errors.
